python_ta/reporters/persistent_server.py

```diff
@@ -0,0 +1,72 @@
+"""This module defines a HTML server using aiohttp that supports reloading via WebSockets."""
+
+import asyncio
+import logging
+import threading
+import webbrowser
+
+from aiohttp import WSMsgType, web
+
+logging.getLogger("aiohttp.access").disabled = True
+latest_html: bytes = b"<h1>Loading report...</h1>"
+websockets = set()
+server_started = False
+loop = asyncio.get_event_loop()
+
+
+async def handle_report(request):
+    """Serve the current HTML content at the root endpoint ('/')."""
+    return web.Response(body=latest_html, content_type="text/html")
+
+
+async def websocket_handler(request):
+    """Handle WebSocket connections, tracking clients and listening for errors."""
+    ws = web.WebSocketResponse()
+    await ws.prepare(request)
+
+    websockets.add(ws)
+
+    try:
+        async for msg in ws:
+            if msg.type == WSMsgType.ERROR:
+                print(f"WebSocket error: {ws.exception()}")
+    finally:
+        websockets.remove(ws)
+    return ws
+
+
+async def update_report(new_html: bytes, port: int):
+    """Update the served HTML content and notify connected WebSocket clients to reload."""
+    global latest_html
+    latest_html = new_html
+
+    active = [ws for ws in websockets if not ws.closed]
+    if active:
+        for ws in active:
+            await ws.send_str("reload")
+    else:
+        webbrowser.open(f"http://localhost:{port}", new=2)
+
+
+def start_server_once(initial_html: bytes, port: int = 8000):
```

Comment:
Don't use a default value, as you're (always) setting the port at the call site of this function